code/modules/projectiles/updated_projectiles/guns/energy.dm

```diff
@@ -371,3 +371,235 @@
 		var/mob/living/carbon/human/user = usr //Hacky...
 		user.update_power_display(perc)
 	return 1
+
+//-------------------------------------------------------
+//Lasguns
+
+/obj/item/weapon/gun/energy/lasgun
+	name = "\improper Lasgun"
+	desc = "A laser based firearm. Uses power cells."
+	origin_tech = "combat=5;materials=4"
+	reload_sound = 'sound/weapons/gun_rifle_reload.ogg'
+	fire_sound = 'sound/weapons/Laser.ogg'
+	matter = list("metal" = 2000)
+	flags_equip_slot = SLOT_BACK
+	w_class = 4
+	force = 15
+	overcharge = 0
+	flags_gun_features = GUN_AUTO_EJECTOR|GUN_CAN_POINTBLANK|GUN_ENERGY
+	aim_slowdown = SLOWDOWN_ADS_RIFLE
+	wield_delay = WIELD_DELAY_SLOW
+	gun_skill_category = GUN_SKILL_RIFLES
+
+
+/obj/item/weapon/gun/energy/lasgun/set_gun_config_values()
+	fire_delay = config.low_fire_delay
+	accuracy_mult = config.base_hit_accuracy_mult + config.max_hit_accuracy_mult
+	accuracy_mult_unwielded = config.base_hit_accuracy_mult - config.high_hit_accuracy_mult
+	damage_mult = config.base_hit_damage_mult
+	scatter_unwielded = config.max_scatter_value * 2 //Heavy and unwieldy
+	damage_falloff_mult = config.med_damage_falloff_mult
+
+
+//-------------------------------------------------------
+//M43 Sunfury Lasgun MK1
+
+/obj/item/weapon/gun/energy/lasgun/M43
+	name = "\improper M43 Sunfury Lasgun MK1"
+	desc = "An accurate, recoilless laser based battle rifle with an integrated charge selector. Ideal for longer range engagements. Uses power cells."
+	force = 15 //Large and hefty!
+	icon_state = "m43"
+	item_state = "m43"
+	attachable_allowed = list(
+						/obj/item/attachable/bayonet,
+						/obj/item/attachable/reddot,
+						/obj/item/attachable/verticalgrip,
+						/obj/item/attachable/angledgrip,
+						/obj/item/attachable/lasersight,
+						/obj/item/attachable/gyro,
```

Comment:
What's with all these attachments?

Reply:
I'm not sure why these would be inappropriate.